src/core/libmaven/jsonReports.cpp

```diff
@@ -1,286 +1,678 @@
-        if( grp.getCompound() == NULL || grp.childCount() == 0 ) {
-            grp.groupId= ++groupId;
-            grp.metaGroupId= ++metaGroupId;
-            if(groupId>1) myfile << "\n,";
-            writeGroupMzEICJson(grp,myfile, vsamples);
-        }
-        else { //output all relevant isotope info otherwise
-            //does this work? is children[0] always the same as grp (parent)?
-            grp.metaGroupId = ++ metaGroupId;
-            for (unsigned int k=0; k < grp.children.size(); k++) {
-                grp.children[k].metaGroupId = grp.metaGroupId;
-                grp.children[k].groupId= ++groupId;
-                if(groupId>1) myfile << "\n,";
-                writeGroupMzEICJson(grp.children[k], myfile, vsamples);
-            }
-        }
-        //Q_EMIT(updateProgressBar("Writing to json file. Please wait...", i, allgroups.size() - 1));
-    }
-    myfile << "]}"; //groups
-    myfile.close();
-    //Q_EMIT(updateProgressBar("Writing to json complete.", 1, 1));
-}
-
-string JSONReports::sanitizeJSONstring(string s) {
-    boost::replace_all(s, "\"","\\\"");
-    //replace(s.begin(),s.end(),"\"","\\\""); //escape quotes
-    s="\""+s+"\""; //quote the whole string
-    return s;
-}
+#include "doctest.h"
+#include "json.hpp"
+
+using json = nlohmann::json;
+
+#include "jsonReports.h"
+#include "Compound.h"
+#include "EIC.h"
+#include "masscutofftype.h"
+#include "mzSample.h"
+#include "mzUtils.h"
+#include "mavenparameters.h"
+#include "datastructures/mzSlice.h"
+#include "databases.h"
+#include "classifierNeuralNet.h"
+#include "PeakDetector.h"
+
+
+JSONReports::JSONReports(MavenParameters* _mp, bool pollyUpload):
+    _uploadToPolly(pollyUpload)
+{
+    _mavenParameters=_mp;
+}
+
+
+void JSONReports::_writeGroup(PeakGroup& grp, ofstream& myfile) {
+    
+    //add labels to json file
+    char lab = grp.label;
+
+    PeakGroup* parentGroup = grp.getParent();
+    if (parentGroup) {
+        if (grp.label == '\0') {
+            lab = parentGroup->label;
+        }
+    }
+
+    myfile << setprecision(10);
+    myfile << "{\n";
+    myfile << "\"groupId\": " << grp.groupId ;
+
+    myfile << ",\n" << "\"label\": ";
+    myfile << "\"";
+    if (lab == 'g' || lab == 'b') myfile << lab;
+    myfile << "\"";
+
+    if(_uploadToPolly) {
+
+        int mlLabel =  (grp.markedGoodByCloudModel) ? 1 : (grp.markedBadByCloudModel) ? -1 : 0;
+        myfile  << ",\n" << "\"ml-label\": " << mlLabel;
+    }
+
+    myfile << ",\n" << "\"metaGroupId\": " << grp.metaGroupId ;
+    myfile << ",\n" << "\"meanMz\": " << grp.meanMz  ;
+    myfile << ",\n" << "\"meanRt\": " << grp.meanRt ;
+    myfile << ",\n" << "\"rtmin\": " << grp.minRt ;
+    myfile << ",\n" << "\"rtmax\": " << grp.maxRt ;
+    myfile << ",\n" << "\"maxQuality\": " << grp.maxQuality ;
+}
+
+
+void JSONReports::_writeCompoundLink(PeakGroup& grp,ofstream& myfile){
+    
+    myfile << setprecision(10);
+    double mz;

```

Comment:
It is a good practice to _always_ initialize any declared variables. That would make this:
```C++
double mz = 0.0f;
```